src/app/Interfaces/Iproduct.ts

```diff
@@ -0,0 +1,15 @@
+export interface Iproducts{
+    products:IProduct[]
```

Comment:
why did make two interfaces?!
if done on "product" you can use this in the component as an array direct.